django/db/models/sql/compiler.py

```diff
@@ -150,7 +147,15 @@ def collapse_group_by(self, expressions, having):
                         getattr(expr.output_field, 'model') == self.query.model):
```

Comment:
I think this approach will add an extraneous expression when a self referential foreign key is involved.

```python
class Person(models.Model):
    best_friend = models.ForeignKey('self', related_name='best_friends')
    age =  models.SmallPositiveIntegerField()

Person.objects.annotate(Avg('best_friends__age'))
```